File: Operators-And-Expressions/PointInACircle/PointInCircle.cs
// Problem 7. Point in a Circle

// Write an expression that checks if given point (x, y) is inside a circle K({0, 0}, 2).

using System;

class PointInCircle
{
    static void Main()
    {
        Console.WriteLine("Insert x: ");
        double x = double.Parse(Console.ReadLine());
        Console.WriteLine("Insert y: ");
        double y = double.Parse(Console.ReadLine());
        Console.WriteLine("Is point (" + x + "," + y + ") inside of the circle?");
        Console.WriteLine((x*x+y*y <= 4) ? "Yes!" : "No!");
    }
}

File: JavaScript Fundamentals Homeworks/Operators and Expressions/point-in-circle.js
// Problem 6. Point in Circle
// Write an expression that checks if given point P(x, y) is within a circle K({0,0}, 5). //{0,0} is the centre and 5 is the radius

var inside = {x:-5, y:0}, outside = {x:-4, y:-3.5};

console.log(isInside(inside.x, inside.y, 0, 0, 5));
console.log(isInside(outside.x, outside.y, 0, 0, 5));

function isInside(x, y, cx, cy, r) {
    return (x - cx) * (x - cx) + (y - cy) * (y - cy) <= r * r;
}